orderer/common/channelparticipation/restapi.go

```diff
@@ -70,13 +70,90 @@ func NewHTTPHandler(config localconfig.ChannelParticipation, registrar ChannelMa
 		router:    mux.NewRouter(),
 	}
 
+	// swagger:operation GET /channels/{channelID} channels listChannel
+	// ---
+	// summary: Returns detailed channel information for an Ordering Service Node (OSN).
+	// parameters:
+	// - name: channelID
+	//   in: path
+	//   description: Channel ID
+	//   required: true
+	//   type: string
+	// responses:
+	//    "200":
+	//       description: Successfully retrieved channel.
+	//       schema:
+	//         "$ref": "#/definitions/channelInfo"
+
 	handler.router.HandleFunc(urlWithChannelIDKey, handler.serveListOne).Methods(http.MethodGet)
 
+	// swagger:operation DELETE /channels/{channelID} channels deleteChannel
+	// ---
+	// summary: Removes an Ordering Service Node (OSN) from a channel.
+	// parameters:
+	// - name: channelID
+	//   in: path
+	//   description: Channel ID
+	//   required: true
+	//   type: string
+	// responses:
+	//    '204':
+	//      description: Successfully removed channel.
+	//    '400':
+	//      description: Bad request.
+	//    '404':
+	//      description: The channel does not exist.
+	//    '405':
+	//      description: The system channel exists, removal is not allowed.
+	//    '409':
+	//      description: The channel is pending removal.
+
 	handler.router.HandleFunc(urlWithChannelIDKey, handler.serveRemove).Methods(http.MethodDelete)
 	handler.router.HandleFunc(urlWithChannelIDKey, handler.serveNotAllowed)
 
+	// swagger:operation GET /channels channels listChannels
+	// ---
+	// summary: Provides a list channel information for an Ordering Service Node (OSN).
+	// responses:
+	//    "200":
+	//       description: Successfully retrieved channel.
+	//       schema:
+	//         "$ref": "#/definitions/channelList"
+
 	handler.router.HandleFunc(URLBaseV1Channels, handler.serveListAll).Methods(http.MethodGet)
 
+	// swagger:operation POST /channels channels createChannel
+	// ---
+	// summary: Joins an Ordering Service Node (OSN) to a channel.
+	// description: If a channel does not yet exist, it will be created.
+	// parameters:
+	// - name: channelID
+	//   in: body
+	//   description: Channel ID
+	//   required: true
+	//   type: string
+	// - name: configBlockPath
+	//   in: body
+	//   description: Path to the file containing an up-to-date config block for the channel.
+	//   required: true
+	//   type: string
```

Comment:
That doesn't look right (I know nothing about swagger, though). The content type is "multipart/form-data", and the one part in the body contains the binary content of the block...

Reply:
It was not right. I removed the channelID param and updated the configBlock param to form data.